lambdas/indexer/app.py

```diff
@@ -334,12 +274,12 @@ class DocumentTally:
     attempts: int
 
     @classmethod
-    def from_message(cls, message) -> 'DocumentTally':
-        body = json.loads(message.body)
+    def from_event_record(cls, record: dict) -> 'DocumentTally':
```

Comment:
```suggestion
    def from_sqs_event(cls, record: dict) -> 'DocumentTally':
```

and read https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#16type-hints

Reply:
The method name has been changed.

 Ended up changing the typing hint to `Dict[str,str]` do you think that it is too generic, or have have another idea for a hint?

Reply:
The event records holds JSON, no? See fourth bullet in the contrib guid section I linked to above.

Reply:
the record would could have JSON present in the `body`, there is also use the 'record ["attributes"]' which contain strings of information.

```
{'attributes': {'ApproximateFirstReceiveTimestamp': '1587423132081',
                'ApproximateReceiveCount': '1',
                'MessageDeduplicationId': 'a4f0df47-3c56-4cae-b908-857be7fd7840',
                'MessageGroupId': '06923068-8202-40c4-a42e-d51070fb574c',
                'SenderId': 'AROARZFZ7W77UV6O5LBHR:azul-indexer-amar-index',
                'SentTimestamp': '1587423122076',
                'SequenceNumber': '18853124392961007616'},
 'awsRegion': 'us-east-1',
 'body': '{"entity_type": "cell_suspensions", "entity_id": '
         '"06923068-8202-40c4-a42e-d51070fb574c", "num_contributions": 1}',
 'eventSource': 'aws:sqs',
 'eventSourceARN': 'arn:aws:sqs:us-east-1:122796619775:azul-documents-amar.fifo',
 'md5OfBody': '40709903053d7dfea1e3fb04b29bb12f',
 'messageAttributes': {},
 'messageId': '22d70fac-7ee4-471c-9377-211fd341c7b6'}
```

For that reason i ~think~ this should work.
   def from_sqs_record(cls, record: Mapping[str, Union[str, JSON]]) -> 'DocumentTally':

However because we know which values contain JSON,  a typing.TypedDict might even be more suitable for this... (certainly is not a "generic" approach)

Thanks for your patience with my questions, I've not used Typing much in the past and i would like learn correct usage + habits going forward.

Reply:
Changed to `Mapping[str, JSON]`